rdt-app/src/main/java/io/ona/rdt_app/contract/PatientRegisterFragmentContract.java

```diff
@@ -1,22 +1,24 @@
 package io.ona.rdt_app.contract;
 
+import android.app.Activity;
+
 import org.json.JSONException;
-import org.json.JSONObject;
+import org.smartregister.view.contract.BaseRegisterFragmentContract;
 
-import io.ona.rdt_app.callback.OnFormSavedCallback;
+import io.ona.rdt_app.model.Patient;
 
 /**
  * Created by Vincent Karuri on 13/06/2019
  */
 public interface PatientRegisterFragmentContract {
 
     interface Presenter {
-        void saveForm(JSONObject jsonForm, OnFormSavedCallback onFormSavedCallback) throws JSONException;
-
         String getMainCondition();
+
+        void launchForm(Activity activity, String formName, Patient patient) throws JSONException;
```

Comment:
Same thing as above, let the presenter handle the exceptions, so that logic is removed from views

Reply:
Done.